From b7552b573853182a228d2d9b12ae79eff8111ee4 Mon Sep 17 00:00:00 2001
From: Ding Wei <leo.ding@rock-chips.com>
Date: Mon, 24 Feb 2020 14:49:26 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3368: fix issue for vdpu_mmu and
 vepu_mmu

tips: vdpu_mmu and vepu_mmu have the same address, than it will double
map when probe. Thus, merge this to vpu_mmu to avoid it.

Change-Id: I0482e3c9a992aa93a1d28ed389ec3ef63ab4a6ee
Signed-off-by: Ding Wei <leo.ding@rock-chips.com>
---
 .../boot/dts/rockchip/rk3368-android.dtsi     |  6 +----
 arch/arm64/boot/dts/rockchip/rk3368.dtsi      | 23 +++++--------------
 2 files changed, 7 insertions(+), 22 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi b/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi
index e41514818c2d..29a9e9d29fb9 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3368-android.dtsi
@@ -202,15 +202,11 @@
 	status = "okay";
 };
 
-&vdpu_mmu {
-	status = "okay";
-};
-
 &vepu {
 	status = "okay";
 };
 
-&vepu_mmu {
+&vpu_mmu {
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3368.dtsi b/arch/arm64/boot/dts/rockchip/rk3368.dtsi
index 040102810141..1bcf18bdff76 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3368.dtsi
@@ -1698,7 +1698,7 @@
 		clock-names = "aclk_vcodec", "hclk_vcodec";
 		resets = <&cru SRST_VIDEO_AXI>, <&cru SRST_VIDEO_AHB>;
 		reset-names = "shared_video_a", "shared_video_h";
-		iommus = <&vepu_mmu>;
+		iommus = <&vpu_mmu>;
 		power-domains = <&power RK3368_PD_VIDEO>;
 		rockchip,srv = <&mpp_srv>;
 		rockchip,taskqueue-node = <0>;
@@ -1715,7 +1715,7 @@
 		clock-names = "aclk_vcodec", "hclk_vcodec";
 		resets = <&cru SRST_VIDEO_AXI>, <&cru SRST_VIDEO_AHB>;
 		reset-names = "shared_video_a", "shared_video_h";
-		iommus = <&vdpu_mmu>;
+		iommus = <&vpu_mmu>;
 		power-domains = <&power RK3368_PD_VIDEO>;
 		rockchip,srv = <&mpp_srv>;
 		rockchip,taskqueue-node = <0>;
@@ -1736,23 +1736,12 @@
 		status = "disabled";
 	};
 
-	vepu_mmu: iommu@ff9a0800 {
+	vpu_mmu: iommu@ff9a0800 {
 		compatible = "rockchip,iommu";
 		reg = <0x0 0xff9a0800 0x0 0x100>;
-		interrupts =  <GIC_SPI 9 IRQ_TYPE_LEVEL_HIGH>;
-		interrupt-names = "vepu_mmu";
-		clocks = <&cru ACLK_VIDEO>, <&cru HCLK_VIDEO>;
-		clock-names = "aclk", "iface";
-		power-domains = <&power RK3368_PD_VIDEO>;
-		#iommu-cells = <0>;
-		status = "disabled";
-	};
-
-	vdpu_mmu: iommu@ff9a0800 {
-		compatible = "rockchip,iommu";
-		reg = <0x0 0xff9a0800 0x0 0x100>;
-		interrupts =  <GIC_SPI 10 IRQ_TYPE_LEVEL_HIGH>;
-		interrupt-names = "vdpu_mmu";
+		interrupts =  <GIC_SPI 9 IRQ_TYPE_LEVEL_HIGH>,
+			<GIC_SPI 10 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "vepu_mmu", "vdpu_mmu";
 		clocks = <&cru ACLK_VIDEO>, <&cru HCLK_VIDEO>;
 		clock-names = "aclk", "iface";
 		power-domains = <&power RK3368_PD_VIDEO>;
-- 
2.35.3

